### DISCUSSION TITLE
Commit 73218b01b25ef7c6278343b78395805b7b3b00db broke the insert-child-before.js scriptlet.

### DISCUSSION THREAD

**Discussion by uploadersoasis** [uBlockO/uBO-Scriptlets#23] [Q&A] [answered]
Commit 73218b01b25ef7c6278343b78395805b7b3b00db broke the insert-child-before.js scriptlet.

I have the following uBlock Origin custom rule which uses insert-child-before:

`multiup.org,multiup.eu##+js(insert-child-before, button[namehost="1fichier.com"], a[id="welcome-message"])`

I repurpose this anchor tag, and change its contents and attributes with other rules which still work, but this tag no longer gets moved to the position I set above using the "insert-child-before" scriptlet.

Reverting to the previous commit by using https://raw.githubusercontent.com/uBlock-user/uBO-Scriptlets/fec2e68c2062f02f304718bc5fb6bb53033655f2/scriptlets.js solves this problem.

My question is why did you change the application of the scriptlet from "interactive" to "loading"?  Is the DOM fully loaded at loading?  If not, that is the problem.

**Comment by uBlock-user** [comment, marked as answer]
My apologies for not seeing this thread till today(Had to set notifications to All Activity). I made that change after gorhill changed this in one of the uBO scriptlets where race condition was becoming an issue. Given the nature of this scriptlet, I will revert. Thanks for bringing it to my attention.

commit - https://github.com/uBlock-user/uBO-Scriptlets/commit/9b200ea9b1ab9ee7b968334df05731bed8ca4912
